analyzer/evmabibackfill/evm_abi_backfill.go

```diff
@@ -210,52 +211,46 @@ func (p *processor) ProcessItem(ctx context.Context, batch *storage.QueryBatch,
 			p.logger.Warn("error processing tx args using abi", "contract address", item.ContractAddr, "err", err)
 			return nil
 		}
-		var abiErrName string
-		var abiErr *abi.Error
-		var abiErrArgs []interface{}
-		var errArgs []*abiEncodedArg
+		batch.Queue(
+			queries.RuntimeTransactionEvmParsedFnUpdate,
+			p.runtime,
+			item.Tx.TxHash,
+			method.RawName,
+			txArgs,
+		)
+		// Parse tx error, if it exists.
 		if item.Tx.TxRevertReason != nil {
 			txrr, err := cleanTxRevertReason(*item.Tx.TxRevertReason)
 			if err != nil {
 				// This is most likely an older tx with a plaintext revert reason, such
 				// as "reverted: Ownable: caller is not the owner". In this case, we do
-				// not parse the error with the abi, but we still update the tx table with
-				// the method and args.
-				batch.Queue(
-					queries.RuntimeTransactionEvmParsedFieldsUpdate,
-					p.runtime,
-					item.Tx.TxHash,
-					method.RawName,
-					txArgs,
-					nil, // error name
-					nil, // error args
-				)
+				// not parse the error with the abi.
+				//
+				// Note: Here and below, we do not need to call `queueIncompatibleTxUpdate`
+				// because the RuntimeTransactionEvmParsedFnUpdate query above will
+				// mark the transaction as processed.
```

Comment:
(This whole comment is informational only, now that I've wrapped my mind around the diff. I think the code is fine as-is; this analyzer is not that large or complex.)

FWIW, I find the new flow harder to follow because there's global-ish state one has to keep in mind (= whether the tx is already marked as processed). This comment is useful (and thank you for it!), but it's even better if we don't need it in the first place.

I'd probably structure it as follows:
1) During the parsing step, only convert `item` into a set of variables (basically lines 213-216 in the red diff). However, make all those variables nilable. If a field cannot be parsed into, set it to nil. Do not generate DB queries (to avoid global state), do not return (to allow the fn parser and err parser to both run).
2) Then update the DB, but not when the variable is nil. Sth like `UPDATE ... SET error_message = COALESCE($6, error_message), ...`.

This does have the downside that we cannot actively override any fields with a `NULL`. But that's not ever needed IIUC.

Reply:
Oh I kind of like that, filed https://github.com/oasisprotocol/nexus/pull/613 as an alternative. LMK what you prefer!

Reply:
:star_struck: it looks nicer than I thought :), I'd say let's go with that!